lucene/core/src/java/org/apache/lucene/index/IndexWriter.java

```diff
@@ -607,6 +633,57 @@ DirectoryReader getReader(boolean applyAllDeletes, boolean writeAllDeletes) thro
           }
         }
       }
+      if (onCommitMerges != null) { // only relevant if we do merge on getReader
+        boolean replaceReaderSuccess = false;
+        try {
+          mergeScheduler.merge(mergeSource, MergeTrigger.GET_READER);
+          onCommitMerges.await(maxCommitMergeWaitMillis, TimeUnit.MILLISECONDS);
```

Comment:
There's no need to check whether we timed out here, since we effectively abort all of the point-in-time merges we created by setting `includeMergeReader` to `false` below, right? I wonder if it would be cleaner to eliminate this AtomicBoolean that is shared with this and the merge threads, and instead using the existing merge abort technique that we have? OTOH IDK how that other mechanism works - is it aborting *all* outstanding merges?

Reply:
we actually never abort these merges. there is no reason why we should do that. we might have done most of the work, we only abort merges if we shutdown our writers. I hope that makes sense?

Reply:
OK, I misunderstood the usage of `includeMergeReader`. I guess the idea is that we let them continue, but keep the original (merging) segments in the reader we're opening, so we do need a separate condition for that